cmd/commands/namespace.go

```diff
@@ -49,7 +49,8 @@ func NamespaceInit(manifests map[string]*core.Manifest, kc *core.KubectlClient)
 		),
 		PreRunE: FlagsValidatorAsCobraRunE(
 			FlagsValidationConjunction(
-				AtMostOneOf("gcr", "dockerhub"),
+				AtMostOneOf("gcr", "dockerhub", "no-secret"),
```

Comment:
I don't think this is correct.
We *do* want to support `--secret my-secret` and say `--gcr` (like before). What this means is that *we* create the secret contents (because of `--gcr` but we store it in a secret name chosen by the user)